pages/components/invoice-card.tsx

```diff
@@ -189,69 +200,121 @@ export default function CheckoutCard({
     }, 2000);
   };
 
+  const handleSendMessage = (pubkeyToOpenChatWith: string) => {
+    let { signIn } = getLocalStorageData();
+    if (!signIn) {
+      alert("You must be signed in to send a message!");
+      return;
+    }
+    router.push({
+      pathname: "/direct-messages",
+      query: { pk: nip19.npubEncode(pubkeyToOpenChatWith) },
+    });
+  };
+
+  const formattedTotalCost = formatWithCommas(totalCost, currency);
+
   return (
     <>
-      <Card className="max-w-[700px]">
-        <CardHeader className="flex justify-center gap-3">
-          <span className="text-xl font-bold">Pay with Lightning</span>
-        </CardHeader>
-        <Divider />
-        <CardBody className="flex flex-col items-center">
-          <DisplayCostBreakdown monetaryInfo={productData} />
-        </CardBody>
-        <CardFooter className="flex flex-col items-center">
-          {!paymentConfirmed ? (
-            <div className="flex flex-col items-center justify-center">
-              {qrCodeUrl ? (
-                <>
-                  <Image
-                    alt="Lightning invoice"
-                    className="object-cover"
-                    src={qrCodeUrl}
-                  />
-                  <div className="flex items-center justify-center">
-                    <p className="text-center">
-                      {invoice.length > 30
-                        ? `${invoice.substring(0, 10)}...${invoice.substring(
-                            invoice.length - 10,
-                            invoice.length,
-                          )}`
-                        : invoice}
-                    </p>
-                    <ClipboardIcon
-                      onClick={handleCopyInvoice}
-                      className={`ml-2 h-4 w-4 cursor-pointer ${
-                        copiedToClipboard ? "hidden" : ""
-                      }`}
-                    />
-                    <CheckIcon
-                      className={`ml-2 h-4 w-4 cursor-pointer ${
-                        copiedToClipboard ? "" : "hidden"
-                      }`}
+      {!paymentCard && (
+        <>
+          <Button
+            type="submit"
+            className={SHOPSTRBUTTONCLASSNAMES + " mt-3"}
+            onClick={() => {
+              handleSendMessage(pubkeyOfProductBeingSold);
+            }}
+            startContent={
+              <EnvelopeIcon className="h-6 w-6 hover:text-yellow-500" />
+            }
+          >
+            Message
+          </Button>
+          <Button
+            type="submit"
+            className={SHOPSTRBUTTONCLASSNAMES + " mt-3"}
+            onClick={() => {
+              let { signIn } = getLocalStorageData();
+              if (!signIn) {
```

Comment:
!getLocalStorageData().signIn could be a way to make this cleaner. Or even turn signedIn to a useState